{"event": "config", "data": {"system": {"arch": "cuda", "sshkey": null, "nodes": [{"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-d001.server.mila.quebec", "local": true}], "self": {"name": "local", "ip": "127.0.0.1", "sshport": 22, "user": "root", "main": true, "hostname": "cn-d001.server.mila.quebec", "local": true}}, "dirs": {"base": "/network/scratch/o/ortizgas/data/milabench", "venv": "/network/scratch/o/ortizgas/data/milabench/venv/torch", "data": "/network/scratch/o/ortizgas/data/milabench/data", "runs": "/network/scratch/o/ortizgas/data/milabench/runs", "extra": "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus", "cache": "/network/scratch/o/ortizgas/data/milabench/cache"}, "group": "rlhf-gpus", "install_group": "torch", "install_variant": "cuda", "run_name": "NVIDIA-A100-SXM4-40GB_rlhf-gpus_192.staging", "enabled": true, "capabilities": {"nodes": 1}, "max_duration": 600, "voir": {"options": {"stop": 60, "interval": "1s"}}, "validation": {"usage": {"gpu_load_threshold": 0.5, "gpu_mem_threshold": 0.5}}, "config_base": "/home/mila/o/ortizgas/CODE/milabench/config", "config_file": "/home/mila/o/ortizgas/CODE/milabench/config/standard.yaml", "definition": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf", "plan": {"method": "njobs", "n": 1}, "tags": ["llm", "multigpu", "rl", "rlhf"], "argv": {"--output_dir": "{milabench_extra}/output", "--model_name_or_path": "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size": 64, "--logging_strategy": "no", "--log_level": "critical", "--bf16": true}, "weight": 1.0, "name": "rlhf-gpus", "tag": ["rlhf-gpus", "0"], "job-number": 0, "devices": [0, 1, 2, 3]}, "pipe": null}
{"event": "meta", "data": {"cpu": {"count": 256, "brand": "AMD EPYC 7742 64-Core Processor"}, "os": {"sysname": "Linux", "nodename": "cn-d001.server.mila.quebec", "release": "5.15.0-1048-nvidia", "version": "#48-Ubuntu SMP Thu Mar 21 18:19:02 UTC 2024", "machine": "x86_64"}, "accelerators": {"arch": "cuda", "gpus": {"0": {"minor_number": "1", "device": 0, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 26.0, "power": 55.396, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "1": {"minor_number": "2", "device": 1, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 26.0, "power": 52.063, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "2": {"minor_number": "3", "device": 2, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 27.0, "power": 55.606, "selection_variable": "CUDA_VISIBLE_DEVICES"}, "3": {"minor_number": "7", "device": 3, "product": "NVIDIA A100-SXM4-40GB", "memory": {"used": 620.875, "total": 40960.0}, "utilization": {"compute": 0.0, "memory": 0}, "temperature": 31.0, "power": 55.003, "selection_variable": "CUDA_VISIBLE_DEVICES"}}}, "date": 1733965021.842336, "milabench": {"tag": "v1.0.0_RC1-13-g7a90b16", "commit": "7a90b1691650232ecd63abded0b7be84bb294c05", "date": "2024-11-08 11:28:56 -0500"}, "pytorch": {"torch": "2.4.0+cu121", "compiler": "GCC 9.3", "cpp": "C++ Version: 201703", "intel": "Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications", "mkl": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "openmp": "OpenMP 201511 (a.k.a. OpenMP 4.5)", "lapack": "LAPACK is enabled (usually provided by MKL)", "nnpack": "NNPACK is enabled", "cpu": "CPU capability usage: AVX2", "build_settings": {"BLAS_INFO": "mkl", "BUILD_TYPE": "Release", "CUDA_VERSION": "12.1", "CUDNN_VERSION": "9.1.0", "CXX_COMPILER": "/opt/rh/devtoolset-9/root/usr/bin/c++", "CXX_FLAGS": "-D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wsuggest-override -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow", "LAPACK_INFO": "mkl", "PERF_WITH_AVX": "1", "PERF_WITH_AVX2": "1", "PERF_WITH_AVX512": "1", "TORCH_VERSION": "2.4.0", "USE_CUDA": "ON", "USE_CUDNN": "ON", "USE_CUSPARSELT": "1", "USE_EXCEPTION_PTR": "1", "USE_GFLAGS": "OFF", "USE_GLOG": "OFF", "USE_GLOO": "ON", "USE_MKL": "ON", "USE_MKLDNN": "ON", "USE_MPI": "OFF", "USE_NCCL": "1", "USE_NNPACK": "ON", "USE_OPENMP": "ON", "USE_ROCM": "OFF", "USE_ROCM_KERNEL_ASSERT": "OFF"}}}, "pipe": null}
{"event": "start", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "192", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733947021.9702754}, "pipe": null}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/cache\n", "pipe": "stdout"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884\n", "pipe": "stderr"}
{"event": "line", "data": "  warnings.warn(\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "line", "data": "Some weights of GPTNeoXForSequenceClassification were not initialized from the model checkpoint at EleutherAI/pythia-160m and are newly initialized: ['score.weight']\n", "pipe": "stderr"}
{"event": "line", "data": "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n", "pipe": "stderr"}
{"event": "data", "data": {"task": "main", "time": 1733947056.3214755, "gpudata": {"0": {"memory": [1719.6875, 40960.0], "load": 0.0, "temperature": 26.0, "power": 63.292}, "1": {"memory": [1743.6875, 40960.0], "load": 0.0, "temperature": 26.0, "power": 60.89}, "2": {"memory": [1863.6875, 40960.0], "load": 0.0, "temperature": 26.0, "power": 65.841}, "3": {"memory": [1647.6875, 40960.0], "load": 0.0, "temperature": 30.0, "power": 62.631}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947056.3214755, "iodata": {"read_count": 116, "write_count": 158, "read_bytes": 433152, "read_time": 6, "write_time": 5, "busy_time": 76}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947056.3214755, "netdata": {"bytes_sent": 3479439934529, "bytes_recv": 4715780810658, "packets_sent": 2869979772, "packets_recv": 4047845912, "errin": 0, "errout": 0, "dropin": 2022, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947056.3214755, "cpudata": {"memory": [58493411328, 1081172426752], "load": 0.0}}, "pipe": "data"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "===training policy===\n", "pipe": "stdout"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "line", "data": "\r  0%|          | 0/22 [00:00<?, ?it/s]", "pipe": "stderr"}
{"event": "data", "data": {"progress": [0, 70], "task": "early_stop"}, "pipe": "data"}
{"event": "line", "data": "No overseer found\n", "pipe": "stdout"}
{"event": "data", "data": {"task": "main", "time": 1733947066.3876312, "gpudata": {"0": {"memory": [9179.6875, 40960.0], "load": 0.38, "temperature": 29.0, "power": 111.057}, "1": {"memory": [8417.6875, 40960.0], "load": 0.4, "temperature": 29.0, "power": 155.541}, "2": {"memory": [8521.6875, 40960.0], "load": 0.32, "temperature": 29.0, "power": 111.098}, "3": {"memory": [8035.6875, 40960.0], "load": 0.58, "temperature": 33.0, "power": 169.908}}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947066.3876312, "iodata": {"read_count": 116, "write_count": 200, "read_bytes": 433152, "read_time": 6, "write_time": 5, "busy_time": 112}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947066.3876312, "netdata": {"bytes_sent": 3479440325638, "bytes_recv": 4715781191219, "packets_sent": 2869980979, "packets_recv": 4047847117, "errin": 0, "errout": 0, "dropin": 2022, "dropout": 0}}, "pipe": "data"}
{"event": "data", "data": {"task": "main", "time": 1733947066.3876312, "cpudata": {"memory": [50152140800, 1081172426752], "load": 4.2}}, "pipe": "data"}
{"event": "line", "data": "[rank0]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 0 has a total capacity of 39.39 GiB of which 536.31 MiB is free. Including non-PyTorch memory, this process has 38.86 GiB memory in use. Of the allocated memory 37.04 GiB is allocated by PyTorch, and 349.98 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "\r  0%|          | 0/22 [00:10<?, ?it/s]\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank3]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 3 has a total capacity of 39.39 GiB of which 822.31 MiB is free. Including non-PyTorch memory, this process has 38.58 GiB memory in use. Of the allocated memory 36.73 GiB is allocated by PyTorch, and 375.92 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank1]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 1 has a total capacity of 39.39 GiB of which 684.31 MiB is free. Including non-PyTorch memory, this process has 38.72 GiB memory in use. Of the allocated memory 36.74 GiB is allocated by PyTorch, and 367.19 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 133, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     main()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py\", line 124, in main\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     trainer.train()\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/trl/trainer/ppov2_trainer.py\", line 465, in train\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     accelerator.backward(loss)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/accelerator.py\", line 2196, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     loss.backward(**kwargs)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/_tensor.py\", line 521, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     torch.autograd.backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/__init__.py\", line 289, in backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     _engine_run_backward(\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:   File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/autograd/graph.py\", line 768, in _engine_run_backward\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]:     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass\n", "pipe": "stderr"}
{"event": "line", "data": "[rank2]: torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 4.36 GiB. GPU 2 has a total capacity of 39.39 GiB of which 4.28 GiB is free. Including non-PyTorch memory, this process has 35.11 GiB memory in use. Of the allocated memory 32.85 GiB is allocated by PyTorch, and 654.66 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)\n", "pipe": "stderr"}
{"event": "line", "data": "[rank0]:[W1211 14:57:50.366565901 ProcessGroupNCCL.cpp:1168] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:57:52.182627 140260253729920 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2840321 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:57:52.183025 140260253729920 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2840322 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "W1211 14:57:52.183154 140260253729920 torch/distributed/elastic/multiprocessing/api.py:858] Sending process 2840323 closing signal SIGTERM\n", "pipe": "stderr"}
{"event": "line", "data": "E1211 14:57:54.829684 140260253729920 torch/distributed/elastic/multiprocessing/api.py:833] failed (exitcode: 1) local_rank: 0 (pid: 2840320) of binary: /network/scratch/o/ortizgas/data/milabench/venv/torch/bin/python\n", "pipe": "stderr"}
{"event": "line", "data": "Traceback (most recent call last):\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/bin/accelerate\", line 8, in <module>\n", "pipe": "stderr"}
{"event": "line", "data": "    sys.exit(main())\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py\", line 48, in main\n", "pipe": "stderr"}
{"event": "line", "data": "    args.func(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 1165, in launch_command\n", "pipe": "stderr"}
{"event": "line", "data": "    multi_gpu_launcher(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/accelerate/commands/launch.py\", line 799, in multi_gpu_launcher\n", "pipe": "stderr"}
{"event": "line", "data": "    distrib_run.run(args)\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/run.py\", line 892, in run\n", "pipe": "stderr"}
{"event": "line", "data": "    elastic_launch(\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 133, in __call__\n", "pipe": "stderr"}
{"event": "line", "data": "    return launch_agent(self._config, self._entrypoint, list(args))\n", "pipe": "stderr"}
{"event": "line", "data": "  File \"/network/scratch/o/ortizgas/data/milabench/venv/torch/lib/python3.10/site-packages/torch/distributed/launcher/api.py\", line 264, in launch_agent\n", "pipe": "stderr"}
{"event": "line", "data": "    raise ChildFailedError(\n", "pipe": "stderr"}
{"event": "line", "data": "torch.distributed.elastic.multiprocessing.errors.ChildFailedError: \n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "line", "data": "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py FAILED\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Failures:\n", "pipe": "stderr"}
{"event": "line", "data": "  <NO_OTHER_FAILURES>\n", "pipe": "stderr"}
{"event": "line", "data": "------------------------------------------------------------\n", "pipe": "stderr"}
{"event": "line", "data": "Root Cause (first observed failure):\n", "pipe": "stderr"}
{"event": "line", "data": "[0]:\n", "pipe": "stderr"}
{"event": "line", "data": "  time      : 2024-12-11_14:57:52\n", "pipe": "stderr"}
{"event": "line", "data": "  host      : cn-d001.server.mila.quebec\n", "pipe": "stderr"}
{"event": "line", "data": "  rank      : 0 (local_rank: 0)\n", "pipe": "stderr"}
{"event": "line", "data": "  exitcode  : 1 (pid: 2840320)\n", "pipe": "stderr"}
{"event": "line", "data": "  error_file: <N/A>\n", "pipe": "stderr"}
{"event": "line", "data": "  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html\n", "pipe": "stderr"}
{"event": "line", "data": "============================================================\n", "pipe": "stderr"}
{"event": "end", "data": {"command": ["/home/mila/o/ortizgas/CODE/milabench/milabench/scripts/activator", "/network/scratch/o/ortizgas/data/milabench/venv/torch", "/network/scratch/o/ortizgas/data/milabench/cache", "accelerate", "launch", "--mixed_precision=bf16", "--dynamo_backend=no", "--machine_rank=0", "--num_machines=1", "--multi_gpu", "--gradient_accumulation_steps=1", "--num_cpu_threads_per_process=4", "--main_process_ip=127.0.0.1", "--main_process_port=29400", "--num_processes=4", "/home/mila/o/ortizgas/CODE/milabench/benchmarks/rlhf/main.py", "--output_dir", "/network/scratch/o/ortizgas/data/milabench/extra/rlhf-gpus/output", "--model_name_or_path", "EleutherAI/pythia-1b-deduped", "--per_device_train_batch_size", "192", "--logging_strategy", "no", "--log_level", "critical", "--bf16"], "time": 1733947075.4728587, "return_code": 1}, "pipe": null}
